internal/rest/dequeue_nodes.go

```diff
@@ -69,6 +69,12 @@ func (rest *RestOperations) DequeueNodes(key string) {
 		}
 		if vs_cache_obj != nil {
 			utils.AviLog.Infof("key: %s, msg: nil model found, this is a vs deletion case", key)
+			// In the case of L7 shared VS, the following condition check makes sure the
+			// VIPs persist over AKO reboot.
+			if lib.IsShardVS(key) && !lib.DisableSync {
+				utils.AviLog.Infof("key: %s, VIPs needs to be retained, not performing vs deletion", key)
+				return
+			}
 			rest.deleteVSOper(vsKey, vs_cache_obj, namespace, key, false, false)
```

Comment:
deleteVSOper internally tries deletes other objects like HTTP PolicySet, Poolgroup, Datascript which are referred in the VS. If we skip calling deleteVSOper, then those objects won't be deleted. This would be a problem. 
Also deleteVSOper can be called from multiple functions, so I think its better to add the check inside deleteVSOper function.